### DISCUSSION TITLE
Type conflict between TARGET=native and Cooja compilation

### DISCUSSION THREAD

**Discussion by rahav** [contiki-ng/contiki-ng#2671] [Q&A] [answered]
Given the following definition and LOG statement, `make TARGET=native` pass compilation:
````
#define TIME_TO_WAIT_FOR_ACK 3 * CLOCK_SECOND

LOG_DBG("reTX CmdMsg in = %u ticks if needed\n", TIME_TO_WAIT_FOR_ACK);
````

In Cooja, however, it would not:
````
controller.c:2411:12: error: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'long int' [-Werror=format=]
````

The challenge is that changing the format specifier to `%lu`, then `make TARGET=native` will fail:
````
controller.c:2411:12: error: format ‘%lu’ expects argument of type ‘long unsigned int’, but argument 2 has type ‘int’ [-Werror=format=]
````

**Comment by pjonsson** [comment, marked as answer]
This is fixed after 4.9 in develop, `LOG_DBG("reTX CmdMsg in %" PRIu64 " ticks if needed\n", TIME_TO_WAIT_FOR_ACK);` will work for native/Cooja regardless of what you're running on.
